pallets/external-validators/src/lib.rs

```diff
@@ -0,0 +1,435 @@
+
+        /// A conversion from account ID to validator ID.
+        ///
+        /// Its cost must be at most one storage read.
+        type ValidatorIdOf: Convert<Self::AccountId, Option<Self::ValidatorId>>;
+
+        /// Validate a user is registered
+        type ValidatorRegistration: ValidatorRegistration<Self::ValidatorId>;
+
+        /// Time used for computing era duration.
+        ///
+        /// It is guaranteed to start being called from the first `on_finalize`. Thus value at
+        /// genesis is not used.
+        type UnixTime: UnixTime;
+
+        /// The weight information of this pallet.
+        type WeightInfo: WeightInfo;
+
+        #[cfg(feature = "runtime-benchmarks")]
+        type Currency: Currency<Self::AccountId>
+            + frame_support::traits::fungible::Balanced<Self::AccountId>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::storage_version(STORAGE_VERSION)]
+    pub struct Pallet<T>(_);
+
+    /// The invulnerable, permissioned collators. This list must be sorted.
+    #[pallet::storage]
+    pub type WhitelistedValidators<T: Config> =
+        StorageValue<_, BoundedVec<T::ValidatorId, T::MaxWhitelistedValidators>, ValueQuery>;
+
+    /// The invulnerable, permissioned collators. This list must be sorted.
+    #[pallet::storage]
+    pub type ExternalValidators<T: Config> =
+        StorageValue<_, BoundedVec<T::ValidatorId, T::MaxExternalValidators>, ValueQuery>;
+
+    /// The invulnerable, permissioned collators. This list must be sorted.
+    #[pallet::storage]
+    pub type SkipExternalValidators<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+    /// The invulnerable, permissioned collators. This list must be sorted.
+    #[pallet::storage]
+    pub type ActiveEra<T: Config> = StorageValue<_, ActiveEraInfo>;
+
+    #[pallet::genesis_config]
+    #[derive(DefaultNoBound)]
+    pub struct GenesisConfig<T: Config> {
+        pub whitelisted_validators: Vec<T::ValidatorId>,
+    }
+
+    #[pallet::genesis_build]
+    impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
+        fn build(&self) {
+            let duplicate_validators = self
+                .whitelisted_validators
+                .iter()
+                // T::ValidatorId does not impl Ord or Hash so we cannot collect into set directly,
+                // but we can check for duplicates if we encode them first.
+                .map(|x| x.encode())
+                .collect::<sp_std::collections::btree_set::BTreeSet<_>>();
+            assert!(
+                duplicate_validators.len() == self.whitelisted_validators.len(),
+                "duplicate validators in genesis."
+            );
+
+            let bounded_validators = BoundedVec::<_, T::MaxWhitelistedValidators>::try_from(
+                self.whitelisted_validators.clone(),
+            )
+            .expect("genesis validators are more than T::MaxWhitelistedValidators");
+
+            <WhitelistedValidators<T>>::put(bounded_validators);
```

Comment:
by default skip external validators is at true right?

Reply:
No, the default is false

Reply:
can we make default be true? I would imagine we want to start without it+

Reply:
in any case I dont think it matters much, but just a thought out loud

Reply:
I will add it to genesis

Reply:
*fixed*